Product Update: Love Cycle (#242)
bump-sh/docs: src/_product-updates/2024-06-15-love-cycle.md

---
title: Love Cycle - June 24
tags: [Improvement]
image: /images/changelog/love-cycle-june-2024.png
---

![love-cycle.png](/images/changelog/love-cycle-june-2024.png)

The last few weeks, we have been busy improving Bump.sh through small enhancements, specific user requests, while globally polishing the product. We call these moments "Love Cycles".

Here is the list of every small change we made.

As always, we love to hear about any features you would like to see added to the product. Send us a message at hello (at) bump.sh!

Improvements: 
- When you are subscribed to an API changelog, the link inside the email notifying about a new change now redirects to the specific change page. 

> - We improved the display of oneOf/anyOf alternatives. When there is a lot of alternatives, we now display the list inside a select instead of a horizontally scrolling tab list.

Reviewer: Ah bon?

Je ne le vois pas sur [cette doc de Memo en prod](https://docs.api.memo.bank/operation/operation-gettransaction#operation-gettransaction-200-body-application-json-source) t'as un lien vers la PR qui introduit ça ?

Author: C'est en train d'être poussé, comme le enhance du "changelog entry" et le retravail des accès hub/doc. La PU sera postée une fois toutes ces PRs validées, mais je préparais le terrain ! 🙏

---
title: Love Cycle - June 24
tags: [Improvement]
image: /images/changelog/love-cycle-june-2024.png
---

![love-cycle.png](/images/changelog/love-cycle-june-2024.png)

The last few weeks, we have been busy improving Bump.sh through small enhancements, specific user requests, while globally polishing the product. We call these moments "Love Cycles".

Here is the list of every small change we made.

As always, we love to hear about any features you would like to see added to the product. Send us a message at hello (at) bump.sh!

Improvements: 
- When you are subscribed to an API changelog, the link inside the email notifying about a new change now redirects to the specific change page. 
- We improved the display of oneOf/anyOf alternatives. When there is a lot of alternatives, we now display the list inside a select instead of a horizontally scrolling tab list.

> - The API/Hub visibility has been reworked, to mitigate the risk of mistakes. The visibility of a documentation inside a private hub is now locked to private.

Reviewer 2: ```suggestion
- The public or private visibility setting options of hubs and their docs was too flexible and confusing. Docs in private hubs are now necessarily private, and cannot be public. Docs in public hubs can be either public or private.
```

> - Single change: we worked on the page to remove duplicated information, and put the focus the deployment date and change id, that can both be useful depending on who reads it. We also put some love on the mobile version.

Reviewer 2: I am still confused about talking about "single change". It might not be the most appropriate vocabulary. 

I don't mean to change everything now, but it seems more relevant to talk about a "changelog entry" which clearly refers to an item in a list of multiple items, rather than a "single change" which would apply to a different context as "something that would happen only once".

Author: That could be better, yes! 
I agree that it's confusing, I left a message talking about it on the PR of the Product Update. (cc @jayk-u )

Reviewer: What about “Single changelog entry”, Or “Changelog event”?

Author: Do we need the "single"? It feels like as we say "Changelog" for the global listing, a "Changelog entry" logically becomes an item of this global listing? Wdyt? 

There's a lot of resources talking about a "Changelog entry"
https://keepachangelog.com/en/1.1.0/
https://docs.gitlab.com/ee/development/changelog.html

Reviewer 3: Do we still want to keep this part in the Love Cycle PU, as it has its own entry in the changelog?

Author: Hey! It's gonna be an enhancement of what we shipped a few weeks ago

Quality of life:

> - While looking at the automatic deployment settings of a resource (documentation or hub), we now display the API key of the organization by default, instead of asking you to create one specific to this scope.

Reviewer 2: ```suggestion
- API keys, that let you configure automatic deployments or use the CLI can be set at the organization level, at a Hub level, or at a doc level. By default, you can always use an API key from a parent resource (for instance, the API key of the Hub or the Org that the doc belongs to). We made it clearer in the UI by displaying the parent resource API key in the Automatic Deployment settings page. It still is possible to create a dedicated key for each resource.
```

> - Deployments page: we removed the tabs allowing to filter between All, Released and Not Released deployments to increase the readability of the page.

Reviewer 2: ```suggestion
- Deployments page: we removed the tabs allowing to filter between All, Released and Not Released deployments to improve the page's readability. It is still possible to filter deployments by State (Enqueued, Deployiong, Deployed, Errored).
```

Fixes:

> - Webhook in documentation now display the generic URL https://webhook.example.com.

Reviewer 2: ```suggestion
- Webhooks in documentation now display the generic URL https://webhook.example.com.
```

- We fixed a visual bug on selects, that could make the user think that multiple items were active.
- We fixed a bug that made the title and description of an more recent change disappear when an user unreleased an older deployment.

> 
> 
> 

Reviewer 2: ```suggestion
```
